elastic/kibana
# Adds additional context to recovered alerts of Infrastructure rules

## Summary

Closes https://github.com/elastic/kibana/issues/143725, https://github.com/elastic/kibana/issues/143726

This PR reads the context variables indexed in Alerts-As-Data for `Infrastructure Rules` and adds it to context when alerts are recovered.

Also, the context variables are now flattened before being indexed in AAD for both of the `Infrastructure Rules`.
<img width="429" alt="Screenshot 2022-11-10 at 12 52 40" src="https://user-images.githubusercontent.com/69037875/201083912-e840d469-3ea0-467e-93ae-63f91e1888d1.png">

The context in notification of recovered alert:
<img width="512" alt="Screenshot 2022-11-10 at 12 54 53" src="https://user-images.githubusercontent.com/69037875/201084316-2d5c0619-fa2e-43ce-86b7-ebf685823b10.png">

## Manual Testing

- Create Inventory rule or Metric Threshold rule
- Select a connector to schedule action on `Recovered` alerts
- Add {{context}} in the action template
- Wait for alerts to be generated
- Wait for alerts to be recovered
- Observe contents of `context` when alerts are recovered
- Contents of `context` should be same as which is created when alerts are fired (this can be verified with action on `Active` alerts)

<img width="563" alt="Screenshot 2022-11-08 at 11 58 30" src="https://user-images.githubusercontent.com/69037875/200547558-49695e32-7db0-4c8d-9ef6-989ef9f7abd5.png">

## The context newly added for recovered alerts with this PR
- `cloud.*`
- `host.*`
   - Excluding:
      - `host.cpu.*`
      - `host.disk.*`
      - `host.network.*`
- `orchestrator.*`
- `container.*`
- `labels.*`
- `tags`

### Checklist
